samcli/lib/init/cli_template_modifier.py

```diff
@@ -0,0 +1,188 @@
+"""
+Class to parse and update template when tracing is enabled
+"""
+import logging
+from yaml.parser import ParserError
+from samcli.yamlhelper import parse_yaml_file
+
+
+LOG = logging.getLogger(__name__)
+
+
+class TemplateModifier:
+    GLOBALS = "Globals:\n"
+    RESOURCES = "Resources:\n"
+    FUNCTION = "  Function:\n"
```

Comment:
Most of the init templates are using 2 spaces for indentation. But yaml has no strict requirement for indentation as long as it's indented by space(s) but not tab(s). 

If we accidentally accept PR that adds an init template with non-2-spaces indentation, will this break?
If so, can we add some form of protective mechanism? e.g. enforcing 2-spaces indentation in the init templates repo?

Reply:
To record the discussions around this topic:
Yes, we discovered several use cases that `Tracing` can't be added successfully. In that case, command will not break but will only output the template without `Tracing`. A warning message will be shown to guide customers to add `Tracing` manually.

We are okay with the current approach for now. But in future if we need extensive support in modifying yaml while preserving comments, we should consider using a yaml library with such support, e.g. raumel.

Reply:
To record the discussions around this topic:
Yes, we discovered several use cases that `Tracing` can't be added successfully. In that case, command will not break but will only output the template without `Tracing`. A warning message will be shown to guide customers to add `Tracing` manually.

We are okay with the current approach for now. But in future if we need extensive support in modifying yaml while preserving comments, we should consider using a yaml library with such support, e.g. raumel.

Reply:
`Tracing` will be added successfully except for 1 situation where the customer's template has weird spacing. In which case it reverts back to the users template with a warning message.